src/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -39,5 +42,37 @@ public void SendAsync_SimpleGet_Success()
                 _output.WriteLine(responseContent);
             }
         }
+
+        [Fact]
+        [OuterLoop]
+        public void SendAsync_SlowServerAndCancel_ThrowsTaskCanceledException()
+        {
+            var handler = new WinHttpHandler();
+            using (var client = new HttpClient(handler))
+            {
+                var cts = new CancellationTokenSource();
+                Task<HttpResponseMessage> t = client.GetAsync(SlowServer, cts.Token);
+
+                Thread.Sleep(500);
```

Comment:
Complete outsider, but you should be able to change the return type to `async Task` and the test framework will take care of the rest. Using `Task.Delay` frees up the thread to do other work (like other tests). Thread.Sleep just blocks it. @stephentoub or @CIPop could elaborate better.